docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,112 @@
+
+3. Implement new logic in estimation chore to suspend nodes
+    - Refactor downtimeTrackingDB method `GetOfflineTime` to check for entries which contain downtime from outside the measurement period
+
+        There is a particular characteristic of downtime tracking to take note of here:<br>
+        
+        A downtime entry tracked at March 2nd 00:00 contains downtime for some length of time preceding that point. See [downtime tracking](storage-node-downtime-tracking.md)<br>
+        
+        If we want to measure total downtime starting at March 1st 00:00, and there is a March 1st 01:00 entry indicating 2 hours of downtime, we need to truncate the measured downtime in our calculation in order to avoid unfairly including downtime which occurred outside the tracking period.<br>
+        
+        We should be able to solve this problem by taking the first entry within the tracking period and checking if the offline time it contains extends beyond the start of the tracking period. If so, truncate it to the start of the tracking period, then sum the rest of the entries as usual.
+
+    - Refactor relevant overlay methods to handle downtime_suspended nodes: `KnownReliable`, `FindNodes`, etc.
+    - Add notification to SN dashboard indicating suspension for downtime.
+
+4. Implement email notifications if node becomes suspended
+
+5. Implement evaluation chore to reinstate and disqualify nodes
+    - Check total downtime for each suspended node where last_contact_success > last_contact_failure. If the total downtime has fallen below the allowed downtime, the node's suspension is lifted, though it still remains under review.
+    - If node.under_review <= now - (grace period + tracking period) it is eligible for evaluation
+    - If the node is eligible for evaluation, we must also ensure that the entire tracking period is accounted for. If last contact was a failure and occurred before the end of the tracking period, we might have a window of downtime which has not yet been recorded. Thus, to measure all downtime for the tracking period, we must also wait until last_contact_success > failure OR last_contact_failure >= end of tracking period.
+
+        When these conditions are met, measurements should take care to include all downtime within the period. In addition to the lower bound issue explained above under step 3, we also have an upper bound issue in this case. We may have an entry which exists beyond the upper bound of the tracking period, yet holds some amount of downtime which occured within it.<br>
+        <br>
+        For example, if we want to measure the total downtime for node A from March 1st 00:00 to March 31st 00:00, and we have an entry from April 1st 01:00 indicating 2 hours of downtime, we must include the one hour of downtime which occurred within the tracking period in our calculation. 
```

Comment:
in this example, does the tracking period end at April 1, 0:00? if not, does this mean we are accounting for downtime across two tracking periods? do we keep records of multiple tracking periods for nodes?